### DISCUSSION TITLE
[documentation] file structure / output

### DISCUSSION THREAD

**Discussion by iway1** [tgreyuk/typedoc-plugin-markdown#376] [Help] [answered]
(Talking about docusaurus) I find it pretty confusing that the documentation for this plugin doesn't say anything about what triggers a new file to be generated. Hooking it up to my project, by default it's just shoving everything into one big file and I have no idea why or how to fix it. For whatever reason, using vanilla typedoc generates multiple files and an intuitive structure for the site, but `typedoc-plugin-markdown` will just default to one big file when passed in the same entry point

Would be helpful to new comers, I think, to include something about what determines the output structure of the generated docs.

As far as I can tell, it seems like providing multiple entry points to the `entryPoints` array is the only way? So I'd need an entry in that array for every single MD page I want generated? is that right?

**Comment by tgreyuk** [comment, marked as answer]
Currently working on fixing this on the next release (not published yet). These are some notes around how it will work (so watch this space). https://github.com/tgreyuk/typedoc-plugin-markdown/blob/next/packages/typedoc-plugin-markdown/documentation/output-file-strategy.md

**Reply by iway1** [reply]
Oh nice that looks great! thanks. This will be a huge addition. I had to do something *extremely hacky* to get the same output you're describing here as an option
